### PR TITLE
Remove action column and replace with overlay div

### PR DESCRIPTION
**Pull request by DrAnax**
Removed the action column, added an overlay div with the same functionality. Player can click anywhere in the div and get the same functionality.

**Minor bug**: the mouseleave event of the div doesn't always fire when moving the mouse towards the bottom of the screen when pointing at the last row of the table

### CODE REVIEW

**Comment by Zamiell** [comment]
make the PR pass travis

**Comment by Zamiell** [comment]
image

looks bugged on the bottom

**Comment by Zamiell** [comment]
the mouseleave event doesn't seem to work consistent. after like 20 seconds of testing i got it to fail a whole bunch of times, maybe try to clean up the code or something?

**Comment by Zamiell** [comment]
additionally, why are you doing this
```
overlay.on("mouseleave", () => {
    overlay.hide();
  });
```
on every single row? isn't that superfluous? shouldn't it be done once?

**Comment by DrAnax** [comment]
Bottom row was firing the mouseover event because it was 1px taller than the rest due to bottom table border. I used outerHeight instead of height which fixes the problem.

**Comment by Zamiell** [comment]
did you fix the visual bug from the screenshot i posted above?

**Comment by DrAnax** [comment]
I believe I did

**Comment by Zamiell** [comment]
its kind of weird to put the icon in the middle, because the user can actually click anywhere in the row to enter, and not necessarily on the button in the middle.

would it look better if the icon was a tooltip under the cursor?

**Comment by DrAnax** [comment]
I'll give it a try

**Comment by DrAnax** [comment]
Didn't have much success with icons, thinking of going with simple text

image

**Comment by Zamiell** [comment]
it should be possible to have a icon tooltip that follows the mouse around

**Comment by DrAnax** [comment]
How's the build failed given I didn't touch client/src/game/ui/ButtonGroup.ts ?

**Comment by Zamiell** [comment]
no idea

**Comment by Zamiell** [comment]
looks like it is failing on master too

**Comment by Zamiell** [comment]
i think it is just a travis bug

**Comment by Zamiell** [comment]
ok, rebase the PR on master now, i fixed it

**Comment by Zamiell** [comment]
just checked out the new PR, looks good

but the icon is still kinda big, maybe reduce it by 50% or so? what do you think

**Comment by DrAnax** [comment]
I wasn't sure how would you like it; therefore I followed the sub optimal
route by creating the cursors in JS although this should really be done
with png files. But having the cursors implemented in the code makes the
tweaking painless, until we get to the desired result.

I'll send a new PR with new icon sizes soon. By the way, Mel said the same
thing. So who am I to argue?

Στις Παρ, 20 Νοε 2020, 9:03 π.μ. ο χρήστης James <notifications@github.com>
έγραψε:

**Comment by Zamiell** [comment]
haha, sounds good

**Comment by DrAnax** [comment]
Made the icons smaller and put a dark red on the ban icon, which appears when table is full.

**Warning**: do not merge with master, there are two bugs at the moment. I made the PR for your viewing pleasure.

Bug #1 (minor): the div is drawn incorrectly when table rows are partially visible:

image

Bug #2 (major): the mousewheel doesn't scroll the table when the mouse is on top of the overlay div.

I'll let you know.

**Comment by DrAnax** [comment]
Ain't simplicity the best?

**Comment by Zamiell** [comment]
i checked it out, looks good!

**Comment by Zamiell** [comment]
i think the "Join" and the "Eye" icons look great.
the "Play" icon is a bit confusing though when presented as just a black triangle.
it kind of looks like a bug.

any ideas for replacing the triangle with something else?

**Comment by Zamiell** [comment]
also, there are just 3 icons right? join/resume/spectate?

**Comment by DrAnax** [comment]
I'm using four icons, I added the ban icon when someone tries to join a full table (in which case a dark red 'ban' font awesome icon is shown and the row is unclickable). The play icon is indeed a bit confusing, since it's a 'return to your table'. Can't this of anything clever at the moment.

If you think the icons sizes are ok, I can now remove the unecessary parts of the code and turn these icons into png files.

**Comment by Zamiell** [comment]
> If you think the icons sizes are ok,

icon sizes look okay to me, padi/timo do you care?

**Comment by Zamiell** [comment]
also it looks like you might need to rebase the branch on master again, cuz it says it has conflicts

**Comment by timotree3** [comment]
> icon sizes look okay to me, padi/timo do you care?
seems good. will this work on mobile?

**Comment by timotree3** [comment]
Just tested on mobile and I can tap on a row to join a game and scroll in the lobby

**Comment by DrAnax** [comment]
Final icons

all_icons

**Comment by Zamiell** [comment]
what about this
https://fontawesome.com/icons/youtube?style=brands
feel free to browse the other font awesome icons to see if you can find anything interesting

**Comment by DrAnax** [comment]
Better?

all_icons

**Comment by padiwik** [comment]
Was it really that confusing before? The new one looks like the youtube logo and nothing like a "play" button. It also no longer matches the style of the others

**Comment by Zamiell** [comment]
the base triangle doesn't look very good. it just looks like a triangle and not a play button.
pick something else from fontawesome to use

**Comment by padiwik** [comment]
Fine put it in a circle, but that's a pretty universal icon for play

**Comment by Zamiell** [comment]
anax do you wanna see if u can put it in a circle?

**Comment by DrAnax** [comment]
This one ?

https://fontawesome.com/icons/play-circle?style=regular

**Comment by Zamiell** [comment]
yes

**Comment by DrAnax** [comment]
all_icons

**Comment by DrAnax** [comment]
Actual sizes

cursor-forbid
cursor-join
play
view

**Comment by Zamiell** [comment]
seems fine

**Comment by Zamiell** [comment]
it looks like you might need to rebase the branch on master again, cuz it says it has conflicts

**Comment by DrAnax** [comment]
I know, on my way

**Comment by DrAnax** [comment]
Please check, I'm not 100% sure I've done everything right.

**Comment by Zamiell** [comment]
yes sir

File: docs/INSTALL.md [hunk 1]
```diff
@@ -315,4 +315,4 @@ Adjust the "certbot" command below according to what domain names you want to re
...
   - Alternatively, if you are actively changing or developing TypeScript code, leave the `webpack-dev-server.sh` script running and go to "http://localhost/?dev". That way, the code will be automatically compiled whenever you change a file, and the page will automatically refresh.
   - You can also go to "http://localhost/?dev&login=test1" to automatically log in as "test1", "http://localhost/?dev&login=test2" to automatically log in as "test2", and so forth. This is useful for testing a bunch of different users in tabs without having to use an incognito window.
-- If you change any CSS, you might also need to run `build_client.sh crit` to re-generate the critical CSS, which is necessary for the content the users see first. The "crit" version takes longer than `build_client.sh`, so you only need to run it once before committing your changes.
+- If you change any CSS, you might also need to run `build_client.sh crit` to re-generate the critical CSS, which is necessary for the content the users see first. The "crit" version takes longer than `build_client.sh`, so you only need to run it once before committing your changes.
```
Comment (Zamiell) [inline docs/INSTALL.md:318]: what did you change here?

Comment (DrAnax) [reply docs/INSTALL.md:318]: I copied it from current master as ti was

Comment (Zamiell) [reply docs/INSTALL.md:318]: well thats why im confused, if its the same then it shouldn't show up in the PR diff

**Comment by Zamiell** [comment]
looks good, thanks a lot for your work

**Pull request merged**